### PR TITLE
Manage of the changed of the password

### PR DESCRIPTION
**Pull request by AxelHumbert**

### CODE REVIEW

File: src/main/java/fr/uge/chargepointconfiguration/user/UserService.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+   */
+  public User updatePassword(ChangePasswordUserDto changePasswordUserDto) {
+    var userDto = getAuthenticatedUser();
+    var id = userDto.id();
+    var user = userRepository.findById(id);
+    if (user == null) {
+      throw new IllegalArgumentException("User not found with this id : " + id);
+    }
+    if (!passwordEncoder.matches(changePasswordUserDto.oldPassword(), user.getPassword())) {
+      throw new IllegalArgumentException("Bad password");
```
Comment (Idhao) [inline src/main/java/fr/uge/chargepointconfiguration/user/UserService.java:33]: TODO explain next time throw a customException to Controller which will be returning an appropriate Http Response

**Review by Idhao — APPROVED** [review]

File: src/front/src/pages/Account.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    const [user, setUser] = useState<any>(null);
+    const [oldPassword, setOldPassword] = useState("");
+    const [password1, setPassword1] = useState("");
+    const [password2, setPassword2]= useState("");
+    const [isButtonDisabled, setIsButtonDisabled] = useState(true);
+    const [errorMessage, setErrorMessage] = useState("");
+    const [successMessage, setSuccessMessage] = useState("");
+
+    useEffect(() => {
+        fetch("../api/user/me")
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/Account.tsx]: ../api ??

File: src/front/src/pages/Account.tsx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+    const validatePassword = (password : string) => {
+        // Regex to check prerequisites of password.
+        const regex = /^(?=.*\d)(?=.*[!@#$%^&*~"'{([-|`_\\)\]}+°£µ§/:;.,?<>])(?=.*[a-z])(?=.*[A-Z]).{8,30}$/;
+        return regex.test(password);
+    };
+
+    const handleButtonClick = () => {
+        if (oldPassword !== "" && password1 === password2) {
+            fetch("../api/user/updatePassword", {
```
Comment (Adrien-COSTANDI) [inline src/front/src/pages/Account.tsx:51]: ../api ??

File: src/front/src/pages/Account.tsx
Comment (Adrien-COSTANDI) [inline src/front/src/pages/Account.tsx:1]: remove the console.log

**Review by Idhao — APPROVED** [review]

**Review by Adrien-COSTANDI — APPROVED** [review]

**Pull request merged**